scripts/apps/search/controllers/SearchMenuController.ts

```diff
@@ -56,11 +57,12 @@ export default function SearchMenuController(
 
     const initActiveProvider = () => {
         if ($location.path() === '/search') {
-            if (!$location.search()._id) {
-                this.activeProvider = this.providers.find(
-                    (provider) => isEqual($location.search(), getSearchParams(provider)),
-                );
-            }
+            this.activeProvider = this.providers.find(
+                (provider) => isEqual(
+                    _.omit($location.search(), ['_id', 'item', 'action']),
```

Comment:
It would be better to pick the values you need from query parameters instead of omitting.

Reply:
> It would be better to pick the values you need from query parameters instead of omitting.

The reason for using omitting here is When we do a single click on the article then **$location.search()** object added one more key that is **_id**. when we do double click on the article then **$location.search()** object added two more keys that are **item** and **action**. so that's why I used omit method to remove the respective key from the **$location.search()** object to compare with the **getSearchParams(provider)** object.

Reply:
Wouldn't it work to pick keys conditionally as well? If you check that `_id` is present, you pick one set of keys, if not - other.